toplevel/native/topmain.ml

```diff
@@ -48,7 +48,7 @@ let prepare ppf =
       Format.fprintf ppf "Uncaught exception: %s\n" (Printexc.to_string x);
       false
 
-let file_argument name =
+let file_argument ?(constructor=(fun x -> Toploop.File x)) name =
```

Comment:
I'm not fond of the optional argument here. Is there a problem with changing the `(name : string)` parameter into an `(input : input)` parameter?

Reply:
changed